classes/class-p4-taxonomy-campaign.php

```diff
@@ -49,15 +49,17 @@ public function add_block( $block_name, $data ) {
 
 			if ( $block_name && $data ) {
 
-				$shortcode = '[shortcake_' . $block_name;
+				$blockcode = '<!-- wp:planet4-blocks/' . $block_name . ' {';
 				if ( $data ) {
+					$param_array = [];
```

Comment:
This is not readable. No need to concatenate parameters.
Pass the parameters here as an assosiative array with the correct type for each value, json encode the array and build the gutenberg block string

Reply:
fixed.